Objects/regions.c

```diff
@@ -250,6 +328,13 @@ int _Py_CheckRegionInvariant(PyThreadState *tstate)
         }
     }
 
```

Comment:
This needs to be called on `error_occurred` case, otherwise we can't reset the invariant.